clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -127,4 +127,27 @@ private EhcacheEntityResponse invoke(EhcacheEntityMessage message) throws Except
       }
     }
   }
+
+  public EhcacheEntityResponse invokeCompleted(EhcacheEntityMessage message) throws Exception {
+    InvokeFuture<EhcacheEntityResponse> result = endpoint.beginInvoke().message(message).ackCompleted().invoke();
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          EhcacheEntityResponse response = result.get();
+          if (Type.FAILURE.equals(response.getType())) {
+            throw ((Failure) response).getCause();
```

Comment:
Then we will be getting an `EntityException` from the platform for this invocation and that is why we have `throws`. This behaves the same way `invoke` behaves and I thought it was a decided way of handling endpoint invokes.